### PR TITLE
[IDB IGNORE] Reworks baking recipes to use yeast and butter, dismantles food.dmi

### PR DESCRIPTION
**Pull request by MistakeNot4892**
### Baking changes
- Bread no longer requires egg.
- Bread no longer randomly contains meat (presumably from the egg/meat protein changes).
- Dough is now 'leavened dough' and requires 1u yeast to make, and has a minimum temperature of 60C to allow for adding the yeast last (and to not overlap with pie crusts, below).
- Buns are made from leavened dough slices.
- Dough made without yeast (with 1u salt instead, to avoid recipe overlap) is 'unleavened dough' and bakes into flatbread.
- Pie crusts exist, and are made with butter, flour, water, salt and sugar. TODO: egg washes.
- Flat leavened dough is used to make pizzas.
- Flat unleavened dough is used to make flatbread.
- Pie crusts are used to make pies.
- Yeast has been added to the various kitchen packs.

### food.dmi and food_ingredients.dmi
- food.dmi and food_ingredients.dmi have been split into dozens of individual icon files in icons/obj/food.
- Kabobs are called skewers and generate their meat from their reagents.
- Junkfood exists under its own abstract type.
- Shark, bear and xenomeat descend from the butchery meat subtype.
- A lot of erroneous use of the meat material (from animal protein) has been changed to egg or cheese.
- There is now a unit test that checks food, drinks and condiments for having valid icons/icon_states.

## Why and what will this PR improve
- food.dmi and food_ingredients.dmi were monolithic files with a lot of unused or overlapping icon states. Splitting it up let me clear out unused ones and move diona roasts into the modpack.
- Bread does not use eggs IRL.
- Yeast and butter now have a purpose and a map presence.
- There is now some granularity to baking that was missing previously.

## Authorship
Myself with discussion on the Pyrelight server.

## Changelog
:cl:
add: Yeast and butter are now available in the kitchen supplies.
tweak: Dough now requires a 60C temperature and does not use egg.
tweak: Flatbread now uses unleavened dough, made without yeast, while pies use pie crust. Check the codex for updated recipes.
/:cl:

### CODE REVIEW

**Comment by MistakeNot4892** [comment]
Still needs testing but the main body of work is done.

**Comment by MistakeNot4892** [comment]
Ended up expanding this into some food icon stuff. Will document when the work is done.

**Comment by MistakeNot4892** [comment]
Depends on #4533

File: code/modules/reagents/chems/chems_ethanol.dm [hunk 1]
```diff
@@ -545,3 +545,14 @@
 
 	glass_name = "kvass"
 	glass_desc = "An alcoholic drink commonly made from bread."
+
+/decl/material/liquid/ethanol/champagne
```
Comment (out-of-phaze) [inline code/modules/reagents/chems/chems_ethanol.dm]: check line 510, this already exists. if this is meant to be an override for the fantasy modpack that cuts out mentions of France/Earth, it should be moved there

**Review by out-of-phaze — DISMISSED** [review]

File: code/modules/reagents/chems/chems_ethanol.dm (hunk 1 above)
Comment (MistakeNot4892) [reply code/modules/reagents/chems/chems_ethanol.dm]: Deleted/merged definitions, added overrides to the fantasy modpack.

**Review by out-of-phaze — APPROVED** [review]
all the baking recipes probably need REAGENT_REPLACE to avoid raw dough taste, but that can be done later

**Pull request merged**